analyses/focal-cn-file-preparation/01-prepare-cn-file.R

```diff
@@ -263,20 +285,73 @@ cnv_no_xy <- cnv_df %>%
 
 # Merge and annotated no X&Y
 autosome_annotated_cn <- process_annotate_overlaps(cnv_df = cnv_no_xy,
-                                                   txdb_exons = tx_exons) %>%
+                                                   txdb_exons = exons_gr) %>%
   # mark possible amplifications in autosomes
   dplyr::mutate(status = dplyr::case_when(
     copy_number > (2 * ploidy) ~ "amplification",
     TRUE ~ status
   ))
 
+#### Wrangle GISTIC data ---------------------------------------------
+
+# Transpose the GISTIC data.frame
+autosome_transposed_gistic_df <- gistic_df %>%
+  tibble::column_to_rownames("Chromosome Arm") %>%
+  t() %>%
+  as.data.frame() %>%
+  dplyr::select(-c("Xp", "Xq", "Yp", "Yq")) %>%
+  tibble::rownames_to_column("biospecimen_id")
+
+# Gather the chromosome arm data into one column and the GISTIC call in another
+autosome_transposed_gistic_df <- autosome_transposed_gistic_df %>%
+  dplyr::mutate(biospecimen_id_copy = biospecimen_id) %>%
```

Comment:
Wait, why are you making a copy?

Reply:
This was the "hacky" thing I implemented to get the `gather` function to ideally represent the data. I am sure there is another way to do so, I can look into it again.

Reply:
There’s a way to do it with gather/spread. I’ll send you a code chunk in a tad.

Reply:
Gather and spread are confusing but they come in handy a lot I find. Sometimes you just have to play with the arguments before you get it as you need it.

Reply:
Okay, so for refiguring, I think the move might be 1) What I have pasted below (which is really all the steps you already had but I think some of the other steps were undoing some stuff).  2) Do an inner join `annotated_cn` data, 3) Split into autosome and sex chromosomes 4) Write each autosome and sex chromosome file. I think if you do that, then you won't have to have two sections like this that repeat themselves. 
```
# Transpose the GISTIC data.frame
transposed_gistic_df <- gistic_df %>%
  tibble::column_to_rownames("Chromosome Arm") %>%
  t() %>%
  as.data.frame() %>%
  tibble::rownames_to_column("biospecimen_id") %>%
  # Gather the chromosome arm data into one column and the GISTIC call in another
  tidyr::gather(key = "arm", value = "gistic_call", -biospecimen_id) %>%
  # Filter the GISTIC data to only the overlapping samples
  dplyr::filter(biospecimen_id %in% autosome_annotated_cn$biospecimen_id) %>%
  # Recode the gistic
  dplyr::mutate(gistic_call = dplyr::case_when(gistic_call == -1 ~ "loss",
                                                gistic_call == 0 ~ "neutral",
                                                gistic_call == 1 ~ "gain",
                                                TRUE ~ "amplification")) 
```